File: submissions/cpp/413.cpp
class Solution {
public:
    int numberOfArithmeticSlices(vector<int> &A) {
        int size = A.size();
        if (size <= 2) return 0;
        int len = 0, num = 0;
        for (int i = 1; i < size - 1; ++i) {
            if (A[i] - A[i - 1] == A[i + 1] - A[i])
                ++len;
            else {
                num += len * (len + 1) / 2;
                len = 0;
            }
        }
        if (len)
            num += len * (len + 1) / 2;
        return num;
    }
};

File: LeetCode/python3/413.py
class Solution:
    def numberOfArithmeticSlices(self, A: List[int]) -> int:
        consecutive, curr, sum = 0, 0, 0
        cul = [0 for i in range(len(A))]
        for i in range(2, len(A)):
            if A[i] - A[i - 1] == A[i - 1] - A[i - 2]:
                consecutive += 1
                curr += consecutive
            else:
                consecutive = 0
                sum += curr
                curr = 0
        return sum if curr == 0 else sum + curr
